src/features/admin/routes/variant/variant-form.tsx

```diff
@@ -0,0 +1,124 @@
+import {
+  Box,
+  Button,
+  Divider,
+  FormControl,
+  IconButton,
+  InputAdornment,
+  Stack,
+  TextField,
+  Typography,
+} from "@mui/material";
+import { useFieldArray, useForm } from "react-hook-form";
+import { ChevronRight, Delete } from "@mui/icons-material";
+import { Variant } from "../../types/table-data";
+import { useState } from "react";
+
+type VariantFormProps = {
+  onSubmit: (variant: Variant) => void;
+  variant?: Variant;
+};
+
+function VariantForm({ variant, onSubmit }: VariantFormProps) {
+  const { register, handleSubmit, control } = useForm({
+    defaultValues: variant ?? {
+      variantGroup: {
+        name: "",
+      },
+      singleVariants: [],
+    },
+  });
+
+  function updateVariant(data: Variant) {
```

Comment:
Ideally for "inline" functions we use `const`.

Reply:
`const updateVariant = (data: Variant) => () => {
    onSubmit(data);
  }`
  
  Like this ?